tests/controller_tests.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+import ethereum.tester
+import ethereum.abi
+import serpent
+import binascii
+import re
+import os
+import sys
+import time
+import random
+from binascii import hexlify
+from ethereum import tester as t
+# t.gas_limit = 4100000
+s = t.state()
+c = s.abi_contract('controller_test.se')
+
+
+def test_setController():
+	print("Testing setController")
+	# set the controller = 0x0 now that we have it
+	print "Setting controller to: %s" % binascii.hexlify(c.address)
+	out = c.setController(binascii.hexlify(c.address))
+	print "setController output: %s" % out
+	assert(out == 1), "setController succeeded"
+	# Test a failure
+	try:
+		raise Exception(c.setController(binascii.hexlify(c.address)))
+	except Exception as exc:
+		#print "exception caught: %s" % exc
+		assert(isinstance(exc, ethereum.tester.TransactionFailed)), "setController should fail when attempted by non-controller address"
+
+
+def test_whitelisting():
+	print("Testing Whitelists")
+	address1 = long(t.a1.encode("hex"), 16)
+	address2 = long(t.a2.encode("hex"), 16)
+	# Test addToWhiteList
+	out = c.addToWhitelist(t.a1)
+	print "addToWhitelist output: %s" % out
+	# Check that address is now Whitelisted:
+	out = c.assertIsWhitelisted(t.a1)
+	assert (out == 1), "assertIsWhitelisted failed."
+	# Check an address that shouldn't be Whitelisted:
+	try:
+		raise Exception(c.assertIsWhitelisted(t.a2))
+	except Exception as exc:
+		assert(isinstance(exc, ethereum.tester.TransactionFailed)), "assertIsWhitelisted should fail when address is not Whitelisted"
+
+	# Test removeFromWhitelist
+	out = c.removeFromWhitelist(t.a1, 0)
+	assert (out == 1), "removeFromWhitelist failed."
+
+
+def test_suicideFunds():
+	print("Testing suicideFunds")
+	# Test with non-controller addresses
+	try:
+		raise Exception(c.suicideFunds(t.a1, sender=t.k1))
+	except Exception as exc:
+		assert(isinstance(exc, ethereum.tester.TransactionFailed)), "suicideFunds should fail when attempted by non-controller address"
+	# Test with another non-controller address  (redundant?)
+	try:
+		raise Exception(c.suicideFunds(t.a2, sender=t.k2))
+	except Exception as exc:
+		assert(isinstance(exc, ethereum.tester.TransactionFailed)), "suicideFunds should fail when attempted by non-controller address"
+	# Test with controller address
```

Comment:
t.k0 is *not* the controller --- it's the _owner_ of the controller.  suicideFunds should *never* be called directly, in fact it's actually impossible to do so unless the controller is calling it which is mistakenly set to the contract itself here due to the inset